internal/featuretoggle/monitoring_stack_feature_toggle.go

```diff
@@ -27,20 +32,89 @@ func (m *MonitoringStackFeatureToggle) Name() string {
 }
 
 func (m *MonitoringStackFeatureToggle) GetFeatureToggleIdentifier() string {
+	// TODO: should this be changed to "MONITORING_STACK"?
 	return "EXPERIMENTAL_FEATURES"
 }
 
+// PreManagerSetupHandle sets up to be done before the manager is created.
 func (m *MonitoringStackFeatureToggle) PreManagerSetupHandle(ctx context.Context) error {
-	// nothing to handle before the manager is setup
 	_ = obov1alpha1.AddToScheme(m.SchemeToUpdate)
 	return nil
 }
 
+// PostManagerSetupHandle uses the manager's cached client and scheme to set up the monitoringStackReconcilerOpt
+// addonReconcilerOpts w.r.t this featureToggleHandler
 func (m *MonitoringStackFeatureToggle) PostManagerSetupHandle(ctx context.Context, mgr manager.Manager) error {
-	// use the manager's cached client and scheme to setup the monitoringStackReconcilerOpt addonReconcilerOpts w.r.t this featureToggleHandler
 	*m.AddonReconcilerOptsToUpdate = append(*m.AddonReconcilerOptsToUpdate, addoncontroller.WithMonitoringStackReconciler{
 		Client: mgr.GetClient(),
 		Scheme: mgr.GetScheme(),
 	})
 	return nil
 }
+
+// Enable is ONLY used for Testing. It adds the GetFeatureToggleIdentifier to the
+// FeatureToggles in the AddonOperator Spec. If the FeatureToggles field is changed,
+// the AddonOperator reconciler exits which triggers the addon operator manager
+// to restart with the new configuration.
+func (m *MonitoringStackFeatureToggle) Enable(ctx context.Context) error {
+	return EnableFeatureToggle(ctx, m.Client, m.GetFeatureToggleIdentifier())
+}
+
+// Disable is ONLY used for Testing. It removes the GetFeatureToggleIdentifier from the
+// FeatureToggles in the AddonOperator Spec if it exists. If the FeatureToggles field is changed,
+// // the AddonOperator reconciler exits which triggers the addon operator manager
+// // to restart with the new configuration.
+func (m *MonitoringStackFeatureToggle) Disable(ctx context.Context) error {
+	return DisableFeatureToggle(ctx, m.Client, m.GetFeatureToggleIdentifier())
+}
+
+// PreClusterCreationSetup is ONLY used for testing. It preforms any set up needed before
+// the test cluster is created.
+func (m *MonitoringStackFeatureToggle) PreClusterCreationSetup(ctx context.Context) error {
+	return nil
+}
+
+// PostClusterCreationSetup is ONLY used for test. It preforms any set up needed after
+// the test cluster is created.
+func (m *MonitoringStackFeatureToggle) PostClusterCreationSetup(ctx context.Context, clusterCreated *dev.Cluster) error {
+	observabilityOperatorCatalogSource, err := renderObservabilityOperatorCatalogSource(ctx, clusterCreated)
+	if err != nil {
+		return fmt.Errorf("failed to render the observability operator catalog source from its template: %w", err)
```

Comment:
All of these methods should be defined, like previously, in a separate `*_testable.go` file to denote that they're dedicated towards the aspect of testing only and serve no runtime significance. 

IMO we should be against the idea of using comments to do so.

Just my two cents: Comments should be, as much as possible, used to only denote the information which the code cannot represent/explain itself.
For explaining rest of the things such as what a function does, or where it's intended to be used, that should be explained by the code itself through aptly descriptive naming of variables/methods or/and objectively enforced by interfaces and encapsulation.

PS: This is the motivation behind my two cents - https://bpoplauschi.github.io/2021/01/20/Clean-Code-Comments-by-Uncle-Bob-part-2.html  :P